helpers.sh

```diff
@@ -0,0 +1,40 @@
+#!/bin/bash
+set -exo pipefail
+
+function host {
+    case ${DOCKER_LOCATION:-localhost} in
+    localhost)
+        echo "localhost"
+        ;;
+    docker_ip)
+        if [[ !  -z  ${DOCKER_HOST}  ]]
+        then
+            DOCKER_IP=`echo ${DOCKER_HOST} | awk -F/ '{print $3}'| awk -F: '{print $1}'`
+        fi
+
+        echo ${DOCKER_IP}
+        ;;
+    container_ip)
+        echo "$(docker inspect -f '{{.NetworkSettings.IPAddress}}' ${1})"
+        ;;
+    esac
+}
+
+function remove_containers {
+    docker rm -fv func-postgres-test 2>/dev/null || true
+    docker rm -fv func-mysql-test 2>/dev/null || true
+    docker rm -fv func-minio-test 2>/dev/null || true
+}
+
+function install_swagger_tool {
```

Comment:
https://github.com/swagger-api/validator-badge seems to outline a way to send a file to a service running a swagger validator. the badge would be a nice addition, but sending our file to a validator service seems much cleaner than us downloading a binary every build, if anything because in theory the service will ensure we keep up to date with the latest swagger spec and here we have to manually keep the binary version up to date.

Reply:
Well, the badge is a good thing, but it's not related to this PR, but what if I want to modify swagger how can know if that is okay?

I can add the badge, but we do know that github cache works really slow, we've seen that couple times with swagger that's why we use huge docker image with swagger tool only to validate our swagger doc through the mounted file:
```
docker run -v /Users/denismakogon/Documents/oracle/go/src/github.com/fnproject/fn:/go/src/github.com/fnproject/fn --rm quay.io/goswagger/swagger validate /go/src/github.com/fnproject/fn/docs/swagger.yml
```
The point of downloading binary is to reduce the time of swagger doc validation.
FYI, the swagger tool image is 493MB, swagger tool binary ~2MB.

Reply:
Not worried about the badge so much here, it would be nice to have separate from this. We can send the file from on disk inside of CI to the swagger server listed from that link for validation (I think), which requires no binary download and then we don't have to worry about versioning of the binary. If we change swagger we had the same issue with the image previously, I don't think this is much of a concern vs us lagging behind swagger updates by manually downloading the binary.

Reply:
Unfortunately, recommended method (posting swagger doc as the payload) doesn't work:
```
curl -X POST -d @docs/swagger.yml -H 'Content-Type:application/json' http://online.swagger.io/validator/debug
<html>
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8"/>
<title>Error 500 Internal Server Error</title>
</head>
<body><h2>HTTP ERROR 500</h2>
<p>Problem accessing /validator/debug. Reason:
<pre>    Internal Server Error</pre></p><hr><i><small>Powered by Jetty://</small></i><hr/>

</body>
</html>
```
so, leaving as is.

Reply:
Ok, next idea: can we build a smaller swagger image with that binary to use instead?